ras_party/controllers/account_controller.py

```diff
@@ -0,0 +1,461 @@
+
+    v = Validator(Exists(*expected))
+    if 'id' in party:
+        # Note: there's not strictly a requirement to be able to pass in a UUID, this is currently supported to
+        # aid with testing.
+        log.debug("'id' in respondent post message. Adding validation rule IsUuid")
+        v.add_rule(IsUuid('id'))
+
+    if not v.validate(party):
+        raise RasError(v.errors, 400)
+
+    iac = request_iac(party['enrolmentCode'])
+    if not iac.get('active'):
+        raise RasError("Enrolment code is not active.", status_code=400)
+
+    existing = query_respondent_by_email(party['emailAddress'], session)
+    if existing:
+        raise RasError("User with email address {} already exists.".format(party['emailAddress']), status_code=400)
+
+    case_context = request_case(party['enrolmentCode'])
+    case_id = case_context['id']
+    business_id = case_context['partyId']
+    collection_exercise_id = case_context['caseGroup']['collectionExerciseId']
+    collection_exercise = request_collection_exercise(collection_exercise_id)
+
+    try:
+        survey_id = collection_exercise['surveyId']
+        survey = request_survey(survey_id)
+        survey_name = survey['longName']
+    except KeyError:
+        raise RasError("There is no survey bound for this user with email address: {}".format(party['emailAddress']))
+
+    b = query_business_by_party_uuid(business_id, session)
+    if not b:
+        msg = "Could not locate business with id '{}' when creating business association.".format(business_id)
+        raise RasError(msg, status_code=404)
+
+    # Chain of enrolment processes
+    translated_party = {
+        'party_uuid': party.get('id') or str(uuid.uuid4()),
+        'email_address': party['emailAddress'],
+        'first_name': party['firstName'],
+        'last_name': party['lastName'],
+        'telephone': party['telephone'],
+        'status': RespondentStatus.CREATED
+    }
+
+    try:
+        #  Create the enrolment respondent-business-survey associations
+        r = Respondent(**translated_party)
+        br = BusinessRespondent(business=b, respondent=r)
+        pending_enrolment = PendingEnrolment(case_id=case_id,
+                                             respondent=r,
+                                             business_id=business_id,
+                                             survey_id=survey_id)
+        Enrolment(business_respondent=br,
+                  survey_id=survey_id,
+                  survey_name=survey_name,
+                  status=EnrolmentStatus.PENDING)
+        session.add(r)
+        session.add(pending_enrolment)
+
+        # Notify the case service of this account being created
+        post_case_event(case_id, r.party_uuid, "RESPONDENT_ACCOUNT_CREATED", "New respondent account created")
+
+        _send_email_verification(r.party_uuid, party['emailAddress'])
+    except (orm.exc.ObjectDeletedError, orm.exc.FlushError, orm.exc.StaleDataError, orm.exc.DetachedInstanceError):
```

Comment:
I think all these exceptions inherit from `SQLAlchemyError` so we could just catch that?